test/clj/rems/db/test_entitlements.clj

```diff
@@ -127,131 +118,143 @@
+             (is (= {applicant #{lic-id1 lic-id2}
+                     member #{lic-id1}}
+                    (:application/accepted-licenses (applications/get-application app-id))))
+
+             (entitlements/process-outbox!)
+
+             (testing "db"
+               (is (= #{[applicant "resource1"] [applicant "resource2"]}
+                      (set (map (juxt :userid :resid) (db/get-entitlements {:application app-id}))))))
+             (testing "POST"
+               (let [add-paths (requests-for-paths server "/add")
+                     ga4gh-paths (requests-for-paths server "/ga4gh")]
+                 (is (= #{{:path "/add" :body [{:application app-id :mail "b@o.b" :resource "resource1" :user "bob"}]}
+                          {:path "/add" :body [{:application app-id :mail "b@o.b" :resource "resource2" :user "bob"}]}}
+                        (set add-paths)))
+                 (is (= 2 (count ga4gh-paths)))
+                 (is (every? is-valid-ga4gh? ga4gh-paths))))))))
 
       (testing "approved application, more accepted licenses generates more entitlements"
-        (with-stub-server server
-          (test-data/command! {:type :application.command/accept-licenses
-                               :application-id app-id
-                               :actor member
-                               :accepted-licenses [lic-id1 lic-id2]}) ; now accept all licenses
-
-          (entitlements/process-outbox!)
-
-          (testing "db"
-            (is (= #{[applicant "resource1"] [applicant "resource2"]
-                     [member "resource1"] [member "resource2"]}
-                   (set (map (juxt :userid :resid) (db/get-entitlements {:application app-id}))))))
-          (testing "POST"
-            (let [add-paths (requests-for-paths server "/add")
-                  ga4gh-paths (requests-for-paths server "/ga4gh")]
-              (is (= #{{:path "/add" :body [{:resource "resource1" :application app-id :user "elsa" :mail "e.l@s.a"}]}
-                       {:path "/add" :body [{:resource "resource2" :application app-id :user "elsa" :mail "e.l@s.a"}]}}
-                     (set add-paths)))
-              (is (= 2 (count ga4gh-paths)))
-              (is (every? is-valid-ga4gh? ga4gh-paths))))))
+        (run-with-server
+         {:status 200}
+         (fn [server]
+           (test-data/command! {:type :application.command/accept-licenses
+                                :application-id app-id
+                                :actor member
+                                :accepted-licenses [lic-id1 lic-id2]}) ; now accept all licenses
+
+           (entitlements/process-outbox!)
+
+           (testing "db"
+             (is (= #{[applicant "resource1"] [applicant "resource2"]
+                      [member "resource1"] [member "resource2"]}
+                    (set (map (juxt :userid :resid) (db/get-entitlements {:application app-id}))))))
+           (testing "POST"
```

Comment:
Maybe same actually with these POSTs, bit ambiquous cause it does not actually POST but checks that the action produced correct POST commands. I did not think of this last time.